Lib/test/test_typing.py

```diff
@@ -1074,6 +1074,26 @@ class CG(PG[T]): pass
         with self.assertRaises(TypeError):
             CG[int](42)
 
+    def test_protocol_defining_init(self):
+        class P(Protocol):
+            x: int
+            def __init__(self, x: int) -> None:
+                self.x = x
+
+        # the protocol itself cannot be instantiated
+        with self.assertRaises(TypeError):
+            P()
```

Comment:
But presumably `P(1)` will work?

Reply:
Great catch. This is a bad test with an even worse comment: this TypeError is occurring because it's missing an argument, not because of the no-instantiation check. I'm inclined to just remove this `assertRaises`. What do you think?

Reply:
Agree that the current test doesn't make much sense. It would be useful though to add a test that asserts that `P.__init__` doesn't get clobbered. For example, you could call `P.__init__(some_object, 1)` directly.

(Also, unrelatedly, this change will need a NEWS entry.)

Reply:
I split up the test into 2:
1. Verify that `__init__` is no longer clobbered (as per your suggestion)
2. Check that concrete subclasses can inherit and use __init__ from a protocol class